Xamarin.Essentials/TextToSpeech/TextToSpeech.android.cs

```diff
@@ -0,0 +1,242 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Android.OS;
+using Android.Speech.Tts;
+using AndroidTextToSpeech = Android.Speech.Tts.TextToSpeech;
+
+namespace Xamarin.Essentials
+{
+    public static partial class TextToSpeech
+    {
+        const int maxSpeechInputLengthDefault = 4000;
+
+        static WeakReference<TextToSpeechImplementation> textToSpeechRef = null;
+        static SemaphoreSlim semaphore;
```

Comment:
that would work totally fine. The semaphore is to stop this:

```
// say everything
var lines = new []{ "line 1", "line 2", "..." };
foreach (var line in lines) {
    TextToSpeech.SpeakAsync(line);
}
```

Since we have a tcs in place that is shared between calls we need to ensure we are wiping the other one out.